### PR TITLE
Scan for directory names as if they were files

### PR DESCRIPTION
**Pull request by pemcg**
This PR implements this RFE: https://bugzilla.redhat.com/show_bug.cgi?id=1684592. It allows for a user to add a directory name to search for in the SSA profile as if it were a file. The presence of the directory is reported in the list of discovered files from a VM.

### CODE REVIEW

**Comment by pemcg** [comment]
@miq-bot add_label wip

**Comment by roliveri** [comment]
@jerryk55 Please review, especially the XFS changes.

File: lib/metadata/util/md5deep.rb [hunk 1]
```diff
@@ -107,6 +107,21 @@ def processDir(path, x, xmlNode)
...
+    if x != "." && x != ".."
+      currDir = File.join(path, x)
+      begin
+        if isDir?(currDir)
+          xmlFileNode = xmlNode.add_element("file", "name" => x, "fqname" => currDir)
+          statHash = {}
+          statHash.merge!(getDirStats(currDir))
+          xmlFileNode.add_attributes(statHash)
+        end
+      rescue Errno::EACCES, RuntimeError, SystemCallError
```
Comment (roliveri) [inline lib/metadata/util/md5deep.rb]: Codeclimate is complaining that you're ignoring these rescued errors.
Maybe you should just log the error at info level. That will quiet codeclimate and may also catch some unexpected behavior.

**Comment by pemcg** [comment]
@miq-bot remove_label wip

**Comment by roliveri** [comment]
@pemcg 
I've released version "0.2.22" of the manageiq-smartstate gem.
You need to make a pull request updating the MIQ Gemfile to access the new gem.

**Comment by pemcg** [comment]
@miq_bot add_label hammer:yes

**Comment by pemcg** [comment]
@miq_bot add_label enhancement

**Comment by pemcg** [comment]
@miq_bot add_label z-stream

**Comment by pemcg** [comment]
@miq_bot add_label hammer/yes

**Pull request merged**